src/ripple/basics/strHex.h

```diff
@@ -62,22 +66,33 @@ charUnHex (char c)
 }
 /** @} */
 
-// NIKB TODO cleanup this function and reduce the need for the many overloads
-//           it has in various places.
-template<class FwdIt>
-std::string strHex (FwdIt first, int size)
+template<class InputIterator>
+std::string strHex(InputIterator begin, InputIterator end)
+{
+    std::string result;
+    result.reserve (std::distance(begin, end));
```

Comment:
There are two hex chars for each byte, so this should be `2*std::distance(begin,end)`